app/src/main/java/com/example/pygmyhippo/admin/DeleteImageFragment.java

```diff
@@ -0,0 +1,229 @@
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.squareup.picasso.Picasso;
+
+import java.util.ArrayList;
+
+public class DeleteImageFragment extends Fragment implements StorageOnCompleteListener<Uri> {
+
+    private AdminDeleteImageBinding binding;
+    private NavController navController;
+    private Account signedInAccount;
+    private Image currentImage;
+    private ImageStorage ImageHandler;
+    private EventDB EventDBhandler;
+    private AccountDB AccountDbHandler;
+
+    private ImageView ImageViewer;
+    private Button DeleteButton;
+
+    /**
+     * @param inflater           The LayoutInflater object that can be used to inflate
+     *                           any views in the fragment,
+     * @param container          If non-null, this is the parent view that the fragment's
+     *                           UI should be attached to.  The fragment should not add the view itself,
+     *                           but this can be used to generate the LayoutParams of the view.
+     * @param savedInstanceState If non-null, this fragment is being re-constructed
+     *                           from a previous saved state as given here.
+     * @return
+     */
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        binding = AdminDeleteImageBinding.inflate(inflater, container, false);
+        View root = binding.getRoot();
+        if (getArguments() != null) {
+            signedInAccount = DeleteImageFragmentArgs.fromBundle(getArguments()).getSignedInAccount();
+            currentImage = DeleteImageFragmentArgs.fromBundle(getArguments()).getImage();
+        }
+        return root;
+    }
+
+    /**
+     * @param view               The View returned by {@link #onCreateView(LayoutInflater, ViewGroup, Bundle)}.
+     * @param savedInstanceState If non-null, this fragment is being re-constructed
+     *                           from a previous saved state as given here.
+     */
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        navController = findNavController(view);
+        AccountDbHandler = new AccountDB();
+        EventDBhandler = new EventDB();
+        ImageHandler = new ImageStorage();
+
+        // set the image
+        ImageViewer = view.findViewById(R.id.a_deleteImage_imageView);
+        ImageHandler.getImageDownloadUrl(currentImage.getUrl(), this);
+
+
+        // set up the back button
+        FloatingActionButton backButton = view.findViewById(R.id.a_deleteImage_backBtn);
+        backButton.setOnClickListener(view1 -> {
+            Log.d("EventFragment", "Back button pressed");
+            navController.popBackStack();
+        });
+
+        // set up the delete button
+        DeleteButton = view.findViewById(R.id.a_deleteImage_deleteBtn);
+        DeleteButton.setOnClickListener(view1 -> {
+            Log.d("EventFragment", "Back button pressed");
+
```

Comment:
Is this back  button logging supposed to be in the delete button?

Reply:
oops